### PR TITLE
Make it build with ghc-9.8

### PR DESCRIPTION
**Pull request by erikd**
# Changelog

```yaml
- description: Make it build with ghc-9.8
  type:
    - maintenance    # not directly related to the code
```

# Context

Make it build with ghc-9.8

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by erikd** [comment]
~Still has an SRP for `cardano-api` so should not be merged yet.~ Fixed.

File: cardano-cli/src/Cardano/CLI/Read.hs [hunk 1]
```diff
@@ -148,6 +148,7 @@ import           Data.Word
 import           GHC.IO.Handle (hClose, hIsSeekable)
 import           GHC.IO.Handle.FD (openFileBlocking)
 import qualified Options.Applicative as Opt
+-- import           Prettyprinter (vsep)
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Read.hs]:
```suggestion
```

File: cardano-cli/src/Cardano/CLI/Types/Errors/GenesisCmdError.hs [hunk 2]
```diff
@@ -17,6 +17,7 @@ import           Cardano.CLI.Types.Errors.StakePoolCmdError
 
 import           Control.Exception (IOException)
 import           Data.Text (Text)
+-- import           Prettyprinter
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Types/Errors/GenesisCmdError.hs]:
```suggestion
```

File: cardano-cli/src/Cardano/CLI/Types/Errors/GovernanceQueryError.hs [hunk 3]
```diff
@@ -7,6 +7,8 @@ import           Cardano.Api.Shelley
 
 import           Ouroboros.Consensus.Cardano.Block (EraMismatch)
 
+-- import           Prettyprinter
+
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Types/Errors/GovernanceQueryError.hs]:
```suggestion
```

File: cardano-cli/src/Cardano/CLI/Types/Errors/QueryCmdLocalStateQueryError.hs [hunk 4]
```diff
@@ -12,6 +12,7 @@ import           Cardano.Api.Pretty
 import           Cardano.CLI.Types.Errors.NodeEraMismatchError
 import           Ouroboros.Consensus.Cardano.Block (EraMismatch (..))
 
+-- import           Prettyprinter ((<+>))
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Types/Errors/QueryCmdLocalStateQueryError.hs]:
```suggestion
```

File: flake.lock
Comment (carbolymer) [inline flake.lock:1]: You need to bump hackage here as well
```bash
nix flake update haskellNix/hackage
```

**Review by carbolymer — COMMENTED** [review]
Please apply comments from @smelc and update hackage index in flake.lock.

File: cardano-cli/src/Cardano/CLI/Parser.hs [hunk 5]
```diff
@@ -114,7 +114,6 @@ readerFromAttoParser p =
 eDNSName :: String -> Either String ByteString
 eDNSName str =
   -- We're using 'Shelley.textToDns' to validate the string.
-  let dnsNameText = Text.pack str
-  in case Shelley.textToDns (Text.length dnsNameText) dnsNameText of
-      Nothing -> Left $ "DNS name is more than 64 bytes: " <> str
-      Just dnsName -> Right . Text.encodeUtf8 . Shelley.dnsToText $ dnsName
+  case Shelley.textToDns 100 (Text.pack str) of
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Parser.hs]: why 100?

Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Parser.hs]: Why not? Few host names are likely to be longer than 100 characters. Happy to use some other value there.

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Parser.hs]: 128 bytes seems to be the [limit](https://github.com/IntersectMBO/cardano-ledger/commit/0dffa089af91722467d57b35e217122b676f7933#diff-2db82f5dbfb4a80856bebe363f5c1fd257cfa32e9214a3307492f4f73163161aR352). Can you also add a comment that says this value comes from the Conway CDDL spec?

File: flake.lock
Comment (erikd) [reply flake.lock:1]: Sorry, what???

```
> nix flake update haskellNix/hackage
error: cannot find flake 'flake:haskellNix/hackage' in the flake registries

> nix flake update haskellNix
error: cannot find flake 'flake:haskellNix' in the flake registries

> nix flake update hackage
error: cannot find flake 'flake:hackage' in the flake registries
```

Comment (erikd) [reply flake.lock:1]: Above did not work for me. Needed:
```
nix flake lock --update-input haskellNix/hackage
```

**Comment by erikd** [comment]
> Please apply comments from @smelc and update hackage index in flake.lock.

Done!

File: cardano-cli/src/Cardano/CLI/Parser.hs (hunk 5 above)
Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Parser.hs]: Done!

**Review by carbolymer — APPROVED** [review]

**Pull request merged**